app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb

```diff
@@ -138,4 +139,20 @@ def allowed_user?
     school_urn.present? || trust_uid.present? || (local_authority_code.present? &&
       (Rails.env.staging? || Rails.env.development? || ALLOWED_LOCAL_AUTHORITIES.include?(local_authority_code)))
   end
+
+  def use_school_group_if_available
+    user = User.find_by(email: identifier)
+    user_trusts = user&.dsi_data&.fetch("trust_uids", [])
+    user_local_authorities = user&.dsi_data&.fetch("la_codes", [])
+    return unless user_trusts&.any? || user_local_authorities&.any?
```

Comment:
My python brain is screaming that `return if ...none?` is clearer, but old habits die hard

Reply:
I think `if there are no X and no Y` is clearer in English than `unless there are any X or any Y`